packages/blocks/src/api/validation/index.js

```diff
@@ -407,20 +407,31 @@ export function isEqualTagAttributePairs( actual, expected, logger = createLogge
 		return false;
 	}
 
-	// Convert tuples to object for ease of lookup
-	const [ actualAttributes, expectedAttributes ] = [ actual, expected ].map( fromPairs );
+	// Convert tuples to object for ease of lookup. Using a `for` loop instead of `Array#map` since it is better
+	// optimized. Expected attributes names are normalized to lower case for ease of case-insensitive comparisons later.
+	const actualAttributes = {};
+	const expectedAttributes = {};
+	for ( let i = 0; i < actual.length; i++ ) {
+		actualAttributes[ actual[ i ][ 0 ] ] = actual[ i ][ 1 ];
+		expectedAttributes[ expected[ i ][ 0 ].toLowerCase() ] = expected[ i ][ 1 ];
+	}
 
 	for ( const name in actualAttributes ) {
 		// As noted above, if missing member in B, assume different
-		if ( ! expectedAttributes.hasOwnProperty( name ) ) {
+		if (
+			! expectedAttributes.hasOwnProperty( name ) &&
+			// Optimization: case-insensitive check deferred for the minority case that the block is invalid.
+			! expectedAttributes.hasOwnProperty( name.toLowerCase() )
+		) {
 			logger.warning( 'Encountered unexpected attribute `%s`.', name );
 			return false;
 		}
 
+		// As noted above, case-insensitive checks deferred for better optimization.
 		const actualValue = actualAttributes[ name ];
-		const expectedValue = expectedAttributes[ name ];
+		const expectedValue = expectedAttributes[ name ] || expectedAttributes[ name.toLowerCase() ];
```

Comment:
One thing that leaves me a bit uncomfortable with this is in whether one or these would be a falsey value, where if the `expectedValue[ name ]` value was explicitly an empty string, we should not want it to consider the fallback `||` here. In trying to come up with a unit test which would fail, I'm unable to. But for the purpose of keeping this maintainable, I'm thinking we should just simplify how we normalize the case. At the very least, I think we should not use a blanket "falsey" check here, but rather condition against `expectedAttributes.hasOwnProperty`.